### PR TITLE
Fixed testing issue with residual directories causing FileExistsError for makedirs()

### LINKED ISSUE #565: entityset_tests/test_serialization.py--makedir() throwing FileExistsError
### Bug/Feature Request Description

Tests in tests/entityset_tests/test_serialization.py (starting @ line 82) throwing FileExistsError from makedirs(path) --> some test(s) leaving existing directories. 

-----
*Issues created here on Github are for bugs or feature requests. For usage questions and questions about errors, please ask on Stack Overflow with the [featuretools](https://stackoverflow.com/questions/tagged/featuretools) tag. Check the [documentation](https://docs.featuretools.com/#get-help) for further guidance on where to ask your question.*

### PR DESCRIPTION
**Pull request by allisonportis**
Fix #565
Resolved by adding fixture that creates the path and makes the directory, yields the path, and removes the path folder in the finalization code.

### CODE REVIEW

File: featuretools/tests/entityset_tests/test_serialization.py [hunk 1]
```diff
@@ -59,12 +59,19 @@ def test_entityset_description(es):
     assert es.metadata.__eq__(_es, deep=True)
 
 
-def test_invalid_formats(es):
+@pytest.fixture
+def path_management():
     path = os.path.join(CACHE, 'es')
+    os.makedirs(path)
```
Comment (rwedge) [inline featuretools/tests/entityset_tests/test_serialization.py]: we have logic here to handle if the path folder already exists

File: featuretools/tests/entityset_tests/test_serialization.py [hunk 2]
```diff
@@ -59,12 +59,22 @@ def test_entityset_description(es):
     assert es.metadata.__eq__(_es, deep=True)
 
 
-def test_invalid_formats(es):
+@pytest.fixture
+def path_management():
     path = os.path.join(CACHE, 'es')
+    try:
+        os.makedirs(path)
+    except (OSError, IOError):
```
Comment (rwedge) [inline featuretools/tests/entityset_tests/test_serialization.py]: We want to ignore the error raised by `os.makedirs` if the folder already exists but let any unexpected errors get raised.  `os.makedirs` raises an OSError in both python 2 and 3 so we should only catch OSError

Comment (allisonportis) [reply featuretools/tests/entityset_tests/test_serialization.py]: I changed it to catch `OSError` and then check the error code of that error to be `errno.EEXIST` but I can't figure out how to order the imports correctly for `import errno`. I checked the pep8 guidelines and it's with the standard library imports but it still fails the make lint. I tried it in a bunch of different specific places as well.

File: featuretools/tests/entityset_tests/test_serialization.py [hunk 3]
```diff
@@ -52,12 +53,23 @@ def test_entityset_description(es):
...
 
 
-def test_invalid_formats(es):
+@pytest.fixture
+def path_management():
     path = os.path.join(CACHE, 'es')
+    try:
+        os.makedirs(path)
+    except OSError as e:
+        if e.errno != errno.EEXIST:
```
Comment (rwedge) [inline featuretools/tests/entityset_tests/test_serialization.py]: could you add a comment describing what `EEXIST` refers to?

**Pull request merged**